### PR TITLE
Switch CA1805 from Info to Hidden based on user feedback

### PR DESCRIPTION
**Pull request by mavasani**
See https://github.com/dotnet/docs/issues/21181 and https://github.com/dotnet/docs/issues/21137 for context

### CODE REVIEW

File: src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/DoNotInitializeUnnecessarily.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ public sealed class DoNotInitializeUnnecessarilyAnalyzer : DiagnosticAnalyzer
             new LocalizableResourceString(nameof(MicrosoftCodeQualityAnalyzersResources.DoNotInitializeUnnecessarilyTitle), MicrosoftCodeQualityAnalyzersResources.ResourceManager, typeof(MicrosoftCodeQualityAnalyzersResources)),
             new LocalizableResourceString(nameof(MicrosoftCodeQualityAnalyzersResources.DoNotInitializeUnnecessarilyMessage), MicrosoftCodeQualityAnalyzersResources.ResourceManager, typeof(MicrosoftCodeQualityAnalyzersResources)),
             DiagnosticCategory.Performance,
-            RuleLevel.IdeSuggestion,
+            RuleLevel.IdeHidden_BulkConfigurable,
```
Comment (mavasani) [inline src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/DoNotInitializeUnnecessarily.cs:23]: This is the only source change.

**Comment by mavasani** [comment]
Tagging @jeffhandley @vatsalyaagrawal @jmarolf - are we still open for taking analyzer changes to .NET 5.0 GA? If so, do we need any shiproom approvals? This is a very trivial fix based on user feedback.

**Comment by jeffhandley** [comment]
5.0 GA bits are quite locked down at this point; I don't think this would be justified for taking it to tactics.  Let's queue this up for the first SDK update after 5.0 GA.

**Comment by mavasani** [comment]
> 5.0 GA bits are quite locked down at this point; I don't think this would be justified for taking it to tactics. Let's queue this up for the first SDK update after 5.0 GA.

Thanks @jeffhandley, that sounds reasonable.

@mmitche @jmarolf - is release/5.0.1xx still the correct branch for first SDK update after 5.0 GA?

**Comment by mmitche** [comment]
@mavasani Yep

File: src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/DoNotInitializeUnnecessarily.cs (hunk 1 above)
Comment (stephentoub) [reply src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/DoNotInitializeUnnecessarily.cs:23]: Thanks

**Review by stephentoub — APPROVED** [review]

**Pull request merged**